docs: add EAS Simulator preview documentation (#48225)
expo/expo: docs/pages/preview/eas-simulator/run-and-control.mdx

## Install a local build with agent-device

For a local iOS simulator `.app` bundle, `install` uploads the build through the controller:

<Terminal
  cmd={[
    '$ eas simulator:exec npx agent-device@latest install com.example.app ./path/to/MyApp.app --platform ios',
    '$ eas simulator:exec npx agent-device@latest open com.example.app --platform ios',
  ]}
/>

Large local builds take longer because the client uploads them from your machine. Prefer `install-from-source` when an EAS artifact URL is available.

For a local Android APK, use the same command with the Android package, APK path, and `--platform android`.

## Use a development build for live changes

Live iteration requires a [development build](/develop/development-builds/introduction/) with `expo-dev-client`. The development build loads JavaScript from Metro instead of relying only on the bundle embedded at build time.

If the machine cannot create the native build locally, configure an EAS profile with both `developmentClient: true` and `ios.simulator: true`:

```json eas.json
{
  "build": {
    "development-simulator": {
      "developmentClient": true,
      "ios": {
        "simulator": true

>       }

Reviewer: should we include apk here too?

Author: Good call to clarify. A profile with `developmentClient: true` already produces an installable Android APK, so adding `android.buildType: "apk"` would be redundant. I updated the text below to make that explicit.

<Terminal
  cmd={[
    '$ eas simulator:exec npx agent-device@latest --help',
    '$ eas simulator:exec npx agent-device@latest help workflow',
  ]}
/>

## Control the app with Argent

For installation and broader controller guidance, see [Argent and Expo](/agents/argent/).

Install the Argent CLI:

<Terminal cmd={['$ npm install --global @swmansion/argent']} />

Start an Argent-backed session with:

<Terminal cmd={['$ eas simulator:start --platform ios --type argent --non-interactive']} />

Use these commands with a session started with `--type argent`. `simulator:exec` supplies `ARGENT_TOOLS_URL` and `ARGENT_AUTH_TOKEN` from **.env.eas-simulator**, so `argent link` is not required for commands invoked this way:

<Terminal
  cmd={[
    '$ eas simulator:exec argent run reinstall-app --udid <udid> --bundleId com.example.app --appPath ./MyApp.app',
    '$ eas simulator:exec argent run <tool> [args...]',
  ]}
/>

Argent uses its own tools and app installation commands. An Argent session does not also provision an agent-device daemon, so do not run agent-device commands against it.

## Use the iOS browser preview

> Supported iOS sessions return a `webPreviewUrl`. Open it in a desktop browser while the session is active.

Reviewer: same

---
title: Run and control an app
sidebar_title: Run and control an app
description: Install an app on EAS Simulator and control it with agent-device, Argent, or the iOS browser preview.
---

import { Terminal } from '~/ui/components/Snippet';
import { Tab, Tabs } from '~/ui/components/Tabs';

> The remote device starts blank. Install a Simulator- or Emulator-compatible build before opening and controlling the app.

Reviewer 2: ✏️ **style** `mechanics.md#capitalization-is-not-emphasis`

"Simulator" and "Emulator" are ordinary nouns here, not the product names "iOS Simulator" and "Android Emulator", so they stay lowercase. Same on get-started.mdx line 66 and troubleshooting.mdx line 27.

```suggestion
The remote device starts blank. Install a simulator- or emulator-compatible build before opening and controlling the app.
```

<!-- docs-review -->

| Inspect a fixed build or capture evidence    | Local release build or EAS simulator build  |
| See current source changes with Fast Refresh | Development build connected to Metro        |
| Use an existing EAS artifact                 | Matching iOS Simulator build or Android APK |

> **warning** A release build contains the JavaScript from build time. Connecting it to Metro does not enable Fast Refresh. Use a development build for live iteration.

## Choose a controller

Choose a controller before starting the session and pass its type explicitly. Use that controller's commands for the rest of the session.

| Controller   | Start option          | Documentation                                  |
| ------------ | --------------------- | ---------------------------------------------- |
| agent-device | `--type agent-device` | [agent-device and Expo](/agents/agent-device/) |
| Argent       | `--type argent`       | [Argent and Expo](/agents/argent/)             |

## Find the app identifier

Read the resolved app configuration before installing a build:

<Terminal cmd={['$ npx expo config --json']} />

> Use `ios.bundleIdentifier` for iOS commands and `android.package` for Android commands. Configure the missing value in **app.json** or your dynamic app config before creating the build.

Reviewer 2: ✏️ **style** `terminology.md#expo-product-and-feature-names`

The file format does not matter here, so name the generic "app config" and link it rather than listing **app.json** and the dynamic form separately.

```suggestion
Use `ios.bundleIdentifier` for iOS commands and `android.package` for Android commands. Configure the missing value in your [app config](/workflow/configuration/) before creating the build.
```

<!-- docs-review -->

## Install an EAS Build artifact

Create an installable artifact for each platform in an EAS Build profile. iOS requires `ios.simulator: true`, and Android requires an [APK build](/build-reference/apk/):

```json eas.json
{
  "build": {
    "remote-device": {
      "ios": {
        "simulator": true
      },
      "android": {
        "buildType": "apk"
      }
    }
  }
}
```

Build the app before starting EAS Simulator. A build can take long enough for an idle simulator session to lose its controller tunnel:

<Terminal
  cmd={[
    '$ npx --yes eas-cli@latest build --platform ios --profile remote-device --non-interactive',
  ]}
/>

You can also find a completed simulator build:

<Terminal
  cmd={[
    '$ npx --yes eas-cli@latest build:list --platform ios --simulator --status finished --json --non-interactive',
  ]}
/>

> After the build is ready, start an agent-device session and use `install-from-source`. The remote VM downloads the artifact directly:

Reviewer 2: ✏️ **style** `prose.md#word-usage`

"VM" is not spelled out anywhere on this page, so expand it on this first mention.

```suggestion
After the build is ready, start an agent-device session and use `install-from-source`. The remote virtual machine (VM) downloads the artifact directly:
```

<!-- docs-review -->

    '$ npx --yes eas-cli@latest simulator:exec npx agent-device@latest help workflow',
  ]}
/>

## Control the app with Argent

For installation and broader controller guidance, see [Argent and Expo](/agents/argent/).

Install the Argent CLI:

<Terminal cmd={['$ npm install --global @swmansion/argent']} />

Start an Argent-backed session with:

<Terminal
  cmd={[
    '$ npx --yes eas-cli@latest simulator:start --platform ios --type argent --non-interactive',
  ]}
/>

Use these commands with a session started with `--type argent`. `simulator:exec` supplies `ARGENT_TOOLS_URL` and `ARGENT_AUTH_TOKEN` from **.env.eas-simulator**, so `argent link` is not required for commands invoked this way:

<Terminal
  cmd={[
    '$ npx --yes eas-cli@latest simulator:exec argent run reinstall-app --udid <udid> --bundleId com.example.app --appPath ./MyApp.app',
    '$ npx --yes eas-cli@latest simulator:exec argent run <tool> [args...]',
  ]}
/>

Argent uses its own tools and app installation commands. An Argent session does not also provision an agent-device daemon, so do not run agent-device commands against it.

## Use the iOS browser preview

> Supported iOS sessions return a `webPreviewUrl`. Open it in a desktop browser while the session is active. It is for the person viewing the simulator; do not open it inside the remote simulator.

Reviewer 2: 💡 **nit** `review-ai-text.md#t5-restatement`

This is the third statement that `webPreviewUrl` is not an app URL, after get-started.mdx line 162 and troubleshooting.mdx line 145; keep the troubleshooting entry and drop the last sentence here.

<!-- docs-review -->
